apps/omg_db/lib/db.ex

```diff
@@ -65,17 +63,41 @@ defmodule OMG.DB do
                       child_top_block_number: 1,
                       get_single_value: 2
 
-  def start_link(args), do: RocksDB.start_link(args)
+  @default_instance_name OMG.DB.Instance.Default
 
-  def child_spec(), do: RocksDB.child_spec()
-  def child_spec(args), do: RocksDB.child_spec(args)
+  def start_link(args) do
+    args
+    |> Keyword.put_new_lazy(:db_path, fn -> Application.fetch_env!(:omg_db, :path) end)
+    |> prepare_args()
+    |> RocksDB.start_link()
+  end
+
+  def child_spec(args \\ []) do
+    args
+    |> Keyword.put_new_lazy(:db_path, fn -> Application.fetch_env!(:omg_db, :path) end)
+    |> prepare_args()
+    |> RocksDB.child_spec()
+  end
 
+  @doc """
+  Initalizes directory for default instance database storage.
+  """
   def init(path) do
-    RocksDB.init(path)
+    init(path, [@default_instance_name])
   end
 
-  def init() do
-    RocksDB.init()
+  @doc """
+  Initializes directories for multiple database instances. The list of `instances` has to contain
+  atoms in the form `OMG.DB.Instance.<InstanceName>`. Instance storage will be created under snake cased
+  `<InstanceName>` subdirectory of `path`.
+  """
+  def init(path, instances) do
+    :ok = Application.put_env(:omg_db, :path, path, persistent: true)
```

Comment:
Not for `omg_db` yet, as it starts as dependent app for a few others.

Reply:
Probably we could consider to make this line in the release task directly instead of here. So we can keep the dirty work on supervisor/release task and keep general modules clean and less coupled.

Another point is that I've seen in test files calling `DB.init` would also:
```
on_exit(fn ->
      Application.put_env(:omg_db, :path, nil)
```

So it might make it even more clear why cleaning if we move this line out. Though the downside would be exposing more lines of codes outside but I feel that is a bit more correct on the boundary of responsibility.

I will let you take the call though. I don't want to be too pushy and ends up people don't want to refactor at all as this is from existing code 😅 also not sure if you agree with me.